check.py

```diff
@@ -357,28 +357,34 @@ def run_binaryen_js_tests():
 
   for s in sorted(os.listdir(os.path.join(options.binaryen_test, 'binaryen.js'))):
     if not s.endswith('.js'): continue
-    print s
-    f = open('a.js', 'w')
-    binaryen_js = open(os.path.join(options.binaryen_bin, 'binaryen.js')).read()
-    f.write(binaryen_js)
-    if NODEJS:
-      f.write(node_test_glue())
     test_path = os.path.join(options.binaryen_test, 'binaryen.js', s)
     test_src = open(test_path).read()
-    f.write(test_src)
-    f.close()
-    def test(engine):
-      cmd = [engine, 'a.js']
+    def test(engine, runner):
+      print runner + ": " + s
+      f = open(os.path.join(options.binaryen_bin, 'a.js'), 'w')
```

Comment:
adding files to `bin/` should be avoided. one option is to clean up those files, but then there's a risk of not reaching the cleanup code. how about copying the js and wasm files into the current dir? if we forget them there, that's less bad than leaving them in the bin dir. Even better would be to do this in a new subdir just for testing, perhaps, like `test/temp/`.

Reply:
The current dir is the project's base directory, where `a.js` was generated before, which wasn't that great either. Hence, a temp directory sounds ideal, will take a look.